### PR TITLE
fix(snap): set snap version using git tags

### PR DESCRIPTION
**Pull request by MonicaisHer**
update snapcraft.yaml for using git tags to set the version

resolve: #454

Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-ui-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

##### Download edgex-ui and snapcraft it:
```
$ git clone https://github.com/MonicaisHer/edgex-ui-go.git
$ cd edgex-ui-go
$ git checkout fix-snap-version
$ snap install snapcraft
$ sudo snap install multipass
$ cd /edgex-ui-go  
$ snapcraft 
```

About 4 minutes, we should see:

```
Snapped edgex-ui_2.0.1-dev.11_amd64.snap
```

### CODE REVIEW

**Comment by bnevis-i** [comment]
recheck

**Comment by bnevis-i** [comment]
@badboy-huaqiao Raising issue in slack #devops channel.

**Comment by MonicaisHer** [comment]
Hello @badboy-huaqiao, could you please add @tonyespy and @siggiskulason as reviewers and wait with the merge until they have finished their review? Alternatively, we could also keep this PR in draft mode until they are done with their review? Thanks!

**Comment by badboy-huaqiao** [comment]
@MonicaisHer ok, looking forward  @tonyespy and @siggiskulason to finish their review.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -49,6 +65,16 @@ parts:
     stage-packages: [libzmq5]
     override-build: |
       go mod tidy
+
+      GIT_VERSION=$(git describe --tags --abbrev=0 | sed 's/v//')
+      if [ -z "$GIT_VERSION" ]; then
+        GIT_VERSION="0.0.0"
+      fi
+
```
Comment (farshidtz) [inline snap/snapcraft.yaml:73]: There are some redundant code for extracting the version.

I suggest the following:
1. Add `snapcraftctl set-version ${GIT_VERSION}` to line 73.
2. Remove the entire `version` part and adopt the version from `web-static` instead.

**Comment by farshidtz** [comment]
@badboy-huaqiao Looks good to me. Please go ahead with merging.

Thanks

**Comment by badboy-huaqiao** [comment]
@farshidtz @MonicaisHer ok, thanks all for your efforts

**Review by badboy-huaqiao — APPROVED** [review]
LGTM

**Pull request merged**